sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -638,6 +645,17 @@ class Analyzer(
           }
           val newAggregateExpressions = a.aggregateExpressions ++ missingAttrs
           a.copy(aggregateExpressions = newAggregateExpressions)
+        case g: Generate =>
+          if (g.outer && !g.join) {
```

Comment:
If `join` is false, `outer` is ignored, so we don't need to correct it, just remove this branch.